imports/plugins/included/product-admin/client/hocs/withProduct.js

```diff
@@ -115,14 +116,24 @@ const wrapComponent = (Comp) => {
    */
   function WithProduct(props) {
     const { history } = props;
+    const [archiveProducts] = useMutation(ARCHIVE_PRODUCTS);
     const [cloneProducts] = useMutation(CLONE_PRODUCTS);
     const [createProductVariant] = useMutation(CREATE_VARIANT);
 
     return (
       <Comp
         onArchiveProduct={async (product, redirectUrl) => {
-          await handleArchiveProduct(product);
-          history.push(redirectUrl);
+          const opaqueProductIds = await getOpaqueIds([{ namespace: "Product", id: product._id }]);
+          const [opaqueShopId] = await getOpaqueIds([{ namespace: "Shop", id: Reaction.getShopId() }]);
+
+          try {
+            await archiveProducts({ variables: { input: { shopId: opaqueShopId, productIds: opaqueProductIds } } });
+            Alerts.toast(i18next.t("productDetailEdit.archiveProductsSuccess"), "success");
+            history.push(redirectUrl);
+          } catch (error) {
+            Alerts.toast(i18next.t("productDetailEdit.archiveProductsFail", { err: error }), "error");
+            throw new ReactionError("server-error", "Unable to archive product");
```

Comment:
Why throw ReactionError from a component?

Reply:
I was just following the pattern we were already using. I guess it doesn't need to be here.